### PR TITLE
Added utils project 

### PR DESCRIPTION
**Pull request by kamil518**
RpcActionHandler class handles action invoking passed within the request body
Adjusted readme and example projects

### CODE REVIEW

File: Grenache.Example.Client/Program.cs [hunk 1]
```diff
@@ -4,38 +4,59 @@
 
 namespace Grenache.Example.Client
```
Comment (vigan-abd) [inline Grenache.Example.Client/Program.cs]: I will come up with a strange request, can we keep indentation to 2 spaces please?

File: Grenache.Example.Client/Program.cs [hunk 2]
```diff
@@ -4,38 +4,59 @@
...
-      var mapRpcRes = await client.Map("rpc_ping", "test map");
-      Console.WriteLine("Mapped Response: " + string.Join(",", mapRpcRes.Select(x => x.Data)));
+            Console.WriteLine("Map Request: rpc_ping test map");
+            var mapRpcRes = await client.Map("rpc_ping", "test map");
+            Console.WriteLine("Mapped Response: " + string.Join(",", mapRpcRes.Select(x => x.Data)));
+        }
     }
-  }
-}
+}
```
Comment (vigan-abd) [inline Grenache.Example.Client/Program.cs]: please configure IDE to append newline to end of each file, we usually follow linux file format

File: Grenache.Utils/Grenache.Utils.csproj [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
```
Comment (vigan-abd) [inline Grenache.Utils/Grenache.Utils.csproj]: hmm afaik we removed newtonsoft in favor of default dotnet libs, can we use those please? see https://github.com/bitfinexcom/grenache-csharp/pull/1/commits/616015265866f2040e61b74e9995948ebd39769a

File: Grenache.Utils/RpcActionHandler.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        {
+            throw new ArgumentException("JSON must include an 'args' property.");
+        }
+        var args = (JArray)argsToken;
+
+        var theMethod = FindMethodInAssembly(action, args);
+
+        if (theMethod == null)
+        {
+            throw new InvalidOperationException($"No suitable method found for action: {action}");
```
Comment (vigan-abd) [inline Grenache.Utils/RpcActionHandler.cs]: in bfx we have ERR_... format for messages, so more like api codes, can we do the same here please? e.g. ERR_ACTION_NOT_FOUND

**Review by vigan-abd — CHANGES_REQUESTED** [review]

File: Grenache.Example.Client/Program.cs (hunk 1 above)
Comment (kamil518) [reply Grenache.Example.Client/Program.cs]: for c# strange indeed. addressed

File: Grenache.Example.Client/Program.cs (hunk 2 above)
Comment (kamil518) [reply Grenache.Example.Client/Program.cs]: done

File: Grenache.Utils/RpcActionHandler.cs (hunk 4 above)
Comment (kamil518) [reply Grenache.Utils/RpcActionHandler.cs]: done

File: Grenache.Utils/Grenache.Utils.csproj (hunk 3 above)
Comment (kamil518) [reply Grenache.Utils/Grenache.Utils.csproj]: done

File: Grenache.Utils/RpcActionHandler.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    {
+      throw new ArgumentException("ERR_JSON_ARGS_REQUIRED: 'args' property is required in the JSON object");
+    }
+
+    // if (!jsonObject.TryGetValue("args", StringComparison.OrdinalIgnoreCase, out JToken argsToken))
+    // {
+    //   throw new ArgumentException("ERR_JSON_ARGS_REQUIRED: 'args' property is required in the JSON object");
+    // }
+
+
```
Comment (vigan-abd) [inline Grenache.Utils/RpcActionHandler.cs]:
```suggestion
```

**Review by vigan-abd — CHANGES_REQUESTED** [review]

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**